include/circt/Dialect/StaticLogic/StaticLogic.td

```diff
@@ -141,8 +141,20 @@ def PipelineWhileOp : Op<StaticLogic_Dialect, "pipeline.while", []> {
   ];
 
   let extraClassDeclaration = [{
+    // Get the condition block.
     Block &getCondBlock() { return condition().front(); }
+
+    // Get the stages block.
     Block &getStagesBlock() { return stages().front(); }
+
+    // Get the terminator of the pipeline.
+    PipelineTerminatorOp getTerminator();
+
+    // Compute the distance between the two stages.
+    uint64_t computeDistance(PipelineStageOp, PipelineStageOp);
+
+    // Compute the initiation interval based on the iter_args def-use chains.
+    uint64_t computeInitiationInterval();
```

Comment:
In the literature, this is usually called the "recurrence-constrained minimum II" (or shorter, RecMII). As long as we don't have resource constraints, II=RecMII, so the name would be fine for now.

Reply:
I can rename to be explicit